src/sql-parser/src/parser.rs

```diff
@@ -3218,18 +3218,46 @@ impl<'a> Parser<'a> {
     fn parse_materialized_view_option_name(
         &mut self,
     ) -> Result<MaterializedViewOptionName, ParserError> {
-        self.expect_keywords(&[ASSERT, NOT, NULL])?;
-        Ok(MaterializedViewOptionName::AssertNotNull)
+        let option = self.expect_one_of_keywords(&[ASSERT, RETAIN])?;
+        let name = match option {
+            ASSERT => {
+                self.expect_keywords(&[NOT, NULL])?;
+                MaterializedViewOptionName::AssertNotNull
+            }
+            RETAIN => {
+                self.expect_keyword(HISTORY)?;
+                MaterializedViewOptionName::RetainHistory
+            }
+            _ => unreachable!(),
+        };
+        Ok(name)
     }
 
     fn parse_materialized_view_option(
         &mut self,
     ) -> Result<MaterializedViewOption<Raw>, ParserError> {
         let name = self.parse_materialized_view_option_name()?;
+
+        if name == MaterializedViewOptionName::RetainHistory {
+            return self.parse_materialized_view_option_retain_history();
+        }
+
         let value = self.parse_optional_option_value()?;
         Ok(MaterializedViewOption { name, value })
     }
 
+    fn parse_materialized_view_option_retain_history(
+        &mut self,
+    ) -> Result<MaterializedViewOption<Raw>, ParserError> {
+        self.expect_token(&Token::Eq)?;
```

Comment:
`=` should be optional for consistency with other options (cf `parse_kafka_broker`).

```suggestion
        let _ = self.consume_token(&Token::Eq);
```